File: 练习/ch21/21.7Calculator(ObjC调用Swift)/Calculator/ViewController.m
//
//  ViewController.m
//  Calculator
//
//  Created by tonymacmini on 14-8-7.
//  Copyright (c) 2014年 tonyguan. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.mainLabel.text = @"0";
    logic = [[CalcLogic alloc] init];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)operandPressed:(id)sender {
    UIButton* btn = (UIButton*)sender;
    self.mainLabel.text =  [logic calculateByTag:btn.tag withMainLabelString:self.mainLabel.text];
}

- (IBAction)equalsPressed:(id)sender {
    UIButton* btn = (UIButton*)sender;
    self.mainLabel.text =  [logic calculateByTag:btn.tag withMainLabelString:self.mainLabel.text];
}

- (IBAction)clearPressed:(id)sender {
    self.mainLabel.text = @"0";
    [logic clean];
}

- (IBAction)decimalPressed:(id)sender {
    if ( [logic doesStringContainDecimal:self.mainLabel.text] == FALSE) {
        NSString *s =  [self.mainLabel.text stringByAppendingString:@"."];
        self.mainLabel.text  =  s;
    }
}

- (IBAction)numberButtonPressed:(id)sender {
    UIButton* btn = (UIButton*)sender;
    self.mainLabel.text = [logic updateMainLabelStringByNumberTag:btn.tag
                                              withMainLabelString:self.mainLabel.text];
    
}

@end

File: 计算器实战/Calculator/Calculator/ViewController.swift
//
//  ViewController.swift
//  Calculator
//
//  Created by NKO on 14-10-28.
//  Copyright (c) 2014年 NKO. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
                            
    @IBOutlet var mainLabel: UILabel!
    
    var logic:CalcLogic!
    
    @IBAction func numberButtonPressend(sender: AnyObject) {
        var btn = sender as UIButton
        mainLabel.text = logic.updateMainLabelStringByNumberTag(tag: btn.tag, mainLabelsString: mainLabel.text)
    }
    
    @IBAction func operandPressed(sender: AnyObject) {
        var btn = sender as UIButton
        mainLabel.text = logic.calculateByTag(btn.tag, mainLablString: mainLabel.text)
    }
    
    @IBAction func decimalPressed(sender: AnyObject) {
        if !logic.doesStringContainDecimal(mainLabel.text) {
            mainLabel.text = mainLabel.text + "."
        }
    }
    
    @IBAction func equalsPressed(sender: AnyObject) {
        var btn = sender as UIButton
        mainLabel.text = logic.calculateByTag(btn.tag, mainLablString: mainLabel.text)
    }
    
    @IBAction func clear(sender: AnyObject) {
        mainLabel.text = "0"
        logic.clean()
    }
    
    @IBOutlet var clearPressed: UIButton
    
    override func viewDidLoad() {
        super.viewDidLoad()
        logic = CalcLogic()
        mainLabel.text = "0"
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        logic = nil
        // Dispose of any resources that can be recreated.
    }


}

